models/enum_type.go

```diff
@@ -10,6 +11,22 @@ type EnumType struct {
 	Values   []interface{}
 }
 
+// Ensure EnumType implements the Equaler interface
+var _ Equaler = EnumType{}
+var _ Equaler = (*EnumType)(nil)
+
+// Equal returns true if two EnumType objects are equal; otherwise false is returned.
+func (self EnumType) Equal(u Equaler) bool {
+	other := u.(EnumType)
```

Comment:
Probably yes. Shall we implement a check for this "cast"? To be able to have `Equal` as an interface method, one needs to have `Equaler` as an argument.